### ISSUE TITLE
role is broken for CentOS 6 and postgresql93-server

### ISSUE THREAD

**Issue by kostyrevaa** [Open-Future-Belgium/PostgreSQL#26]
```
NOTIFIED: [patrik.uytterhoeven.PostgreSQL-For-RHEL6x | restart postgresql] **** 
failed: [192.168.1.17] => {"failed": true}
msg: Stopping postgresql-9.3 service: [  OK  ]
Starting postgresql-9.3 service: [FAILED]
```
```
# grep FATAL pgstartup.log 
2015-07-18 21:40:00 UTC FATAL:  could not create lock file "/var/run/postgresql/.s.PGSQL.5432.lock": No such file or directory
```

because template renders 
```
# grep -i unix_socket postgresql.conf 
unix_socket_directories = '/tmp,/var/run/postgresql/'
```
and there's no such dir
```
# ll /var/run/postgresql
ls: cannot access /var/run/postgresql: No such file or directory
```

I just can't take ansible seriously because of that stuff.
Seems nobody cares at all about testing and such.
Every single role that I tried to use from galaxy had fatal problems. Sigh.

Yes, I can just to redefine postgresql_unix_socket_directories but come on - isn't it suppose to work by default?

I would suggest you to use test-kitchen or something.

**Comment by kostyrevaa** [comment]
But thank you for sharing anyway for except this it is a great role!

**Comment by Trikke76** [comment]
you could easly fix it for your setup by defining the var postgresql_external_pid_file: for your host 
for ex in /tmp
thx for letting know it's not working anymore on rhel 6

**Comment by Trikke76** [comment]
should work with fix #28

**Comment by kostyrevaa** [comment]
How fixes in #28 that are directed on CentOS 7 suppose to fix what is broken for CentOS 6?
Yes, it's probably working for CentOS7 but that was never an issue.
And what you mean by that "should work"? Testing anyone?

**Comment by Trikke76** [comment]
Archhh sorry my mistake this comment was for another project

I will fix this issue when i have some free time bc at the moment i run everything on rhel 7 only and busy with other projects then postresql.
Not sure that your fix will not break it for rhel7 with selinux activated

**Comment by nirving-versent** [comment]
Works for that using postgresql_external_pid_file.

**Comment by kostyrevaa** [comment]
did you even try it before posting tip about postgresql_external_pid_file ?
It's not working. Period.
setting alone external_pid_file like this
```
external_pid_file: '/var/run/postgresql-9.3.pid'
```
produces
```
#  egrep '(unix|external)' /var/lib/pgsql/9.3/data/postgresql.conf 
external_pid_file = '/var/run/postgresql-9.3.pid'
unix_socket_directories = '/tmp,/var/run/postgresql/'

```
and gives
```
2015-08-18 22:37:46 UTC FATAL:  could not create lock file "/var/run/postgresql/.s.PGSQL.5432.lock": No such file or directory
```
suprise!

and only - I repeat "ONLY" - setting
```
postgresql_unix_socket_directories:
    - /tmp
```
produces
```
#  egrep '(unix|external)' /var/lib/pgsql/9.3/data/postgresql.conf 
external_pid_file = '/var/run/9.3-data.pid'
unix_socket_directories = '/tmp'
```
and starts service as expected
BUT with freaking warning
```
postmaster: could not write external PID file "/var/run/9.3-data.pid": Permission denied

```

**Comment by Trikke76** [comment]
fixed by #34
